pkg/kubelet/kubelet.go

```diff
@@ -1458,6 +1458,15 @@ func (kl *Kubelet) syncPod(pod *api.Pod, mirrorPod *api.Pod, runningPod kubecont
 		return err
 	}
 
+	// Create Mirror Pod for Static Pod if it doesn't already exist
+	if kubepod.IsStaticPod(pod) && mirrorPod == nil {
+		glog.V(3).Infof("Creating a mirror pod for static pod %q", podFullName)
+		if err := kl.podManager.CreateMirrorPod(pod); err != nil {
+			glog.Errorf("Failed creating a mirror pod %q: %v", podFullName, err)
+		}
+		mirrorPod, _ = kl.podManager.GetMirrorPodByPod(pod)
+	}
+
```

Comment:
Creating a pod is different than updating the status. We only need to pod spec to create the mirror pod, and pod spec should not change during the sync. I think it's fine to remove them.